packages/map-template/src/components/ViewSelector/ViewSelector.jsx

```diff
@@ -0,0 +1,226 @@
+
+    // Extract only necessary building properties to create a simplified data structure, memoize the result
+    const buildings = useMemo(() => {
+        return buildingsData.map(building => ({
+            id: building.id,
+            name: building.buildingInfo.name,
+        }));
+    }, [buildingsData]);
+
+    // Handle building click event - uses useCallback to maintain a stable reference
+    // This prevents creating new function instances on each render when passed to BuildingList items
+    const handleBuildingClick = useCallback((buildingId) => {
+        mapsIndoorsInstance.fitBuilding(buildingId);
+        setIsExpanded(false);
+    }, [mapsIndoorsInstance]);
+
+    // Add effect for handling outside clicks on desktop only
+    useEffect(() => {
+        // Only add listener when expanded and on desktop
+        if (!isExpanded || !isDesktop) return;
+
+        function handleClickOutside(event) {
+            // Check if click was outside the dropdown and not on the toggle button
+            if (desktopDropdownRef.current &&
+                !desktopDropdownRef.current.contains(event.target) &&
+                !event.target.closest(toggleButtonRef.current)) {
+                setIsExpanded(false);
+            }
+        }
+
+        document.addEventListener('mousedown', handleClickOutside);
+
+        // Clean up
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [isExpanded, isDesktop]);
+
+    // Early return if the current venue has one building
+    if (buildings.length === 1) {
+        return null;
+    }
+
+    /**
+     * Toggle button component that renders different content based on the isDesktop prop
+     * @param {boolean} props.isDesktop Whether the component is being rendered on desktop
+     */
+    const ToggleButton = () => {
+        /* Render mobile list toggle button if isDesktop is false */
+        if (!isDesktop) {
+            return (
+                <button ref={toggleButtonRef} className="view-selector-toggle-button" onClick={() => setIsExpanded(!isExpanded)}>
+                    <PanViewIcon />
+                </button>
+            );
+        }
+
+        /* Render desktop list toggle button if isDesktop is true */
+        return (
+            <button className="view-selector-toggle-button" onClick={() => setIsExpanded(!isExpanded)}>
+                <PanViewIcon />
+                <span> {t('Pan Map to view')}</span>
+                {isExpanded ? <ChevronDownIcon /> : <ChevronUpIcon />}
+            </button>
+        );
+    };
+
+    /**
+     * Render a list of buildings for the current venue
+     */
+    const BuildingList = () => {
+        return (
+            <div className="building-list">
+                {buildings.map(building => (
+                    <button
+                        key={building.id}
+                        className="building-list-item"
```

Comment:
Let's keep using the BEM principle until we decide otherwise 🙏 

```suggestion
                        className="building-list__item"
```